pkg/kubelet/prober/worker.go

```diff
@@ -316,6 +325,29 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 		return true
 	}
 
+	if resultChanged {
+		var eventType string
+		var eventReason string
+		if result == results.Success {
+			eventType = v1.EventTypeNormal
+			eventReason = events.ContainerProbeResultSuccess
+		} else {
+			eventType = v1.EventTypeWarning
+			eventReason = events.ContainerProbeResultFailure
+		}
+		RecordContainerEvent(
+			w.recorder,
+			w.pod,
+			&w.container,
+			eventType,
+			eventReason,
+			"%s probe result changed to %s%s",
```

Comment:
wasn't the original goal to provide more information about failures being ignored due to the FailureThreshold? I might be missing something, but this Event really feels like the `prober_probe_total` metrics we have in kubelet to me: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/prober/prober_manager.go#L35-L49

Reply:
the idea was to tell when exactly failing probes turned into the change of probe status. i.e. probe keeps flaking and we are getting the noise of these events. But never actually exceed the threshold of failures. And then once we did exceed it. How would one find out about it?

Reply:
I see. Then it is indeed not possible today with the existing metrics, but we could introduce a `prober_probe` gauge that would only be there to expose the current status of each probe.

Reply:
I am fine with either solutions, but I think metrics would be easier to use for the following reason:

1. Probe fail
2. Emit `ContainerProbeResultFailure` event
3. Probe succeed again
4. Emit `ContainerProbeResultSuccess` event
5. Probe starts failing again
6. Emit same event as 2)
7. event from 6) is aggregated with event from 2) and its count is set to 2

The fact that `ContainerProbeResultSuccess` still exists and that `ContainerProbeResultFailure` was updated with a new timestamp and new count instead of having a new Event object might be misleading to consumer backends.

I don't know exactly what is the state of alerting on Events today, but this might be something complex to alert on whereas if you use a metric, the value is either 0 or 1 so it is pretty easy to write an alert with it.

Reply:
I am definitely not an expert with regard to investigating that kind of issues so I will trust your judgement. I just wanted to make sure that we weren't overlooking anything and that we are not introducing something that might again result in an explosion of number of events emitted by kubelet.